sql/plan/alter_index.go

```diff
@@ -283,26 +290,74 @@ func (p *AlterIndex) RowIter(ctx *sql.Context, row sql.Row) (sql.RowIter, error)
 	return sql.RowsToRowIter(sql.NewRow(types.NewOkResult(0))), nil
 }
 
-// WithChildren implements the Node interface.
-func (p *AlterIndex) WithChildren(children ...sql.Node) (sql.Node, error) {
+// For plan.AlterIndex, one child is expected - The table.
+func (p AlterIndex) WithChildren(children ...sql.Node) (sql.Node, error) {
 	if len(children) != 1 {
 		return nil, sql.ErrInvalidChildrenNumber.New(p, len(children), 1)
 	}
 
 	switch p.Action {
-	case IndexAction_Create:
-		return NewAlterCreateIndex(p.db, children[0], p.IndexName, p.Using, p.Constraint, p.Columns, p.Comment), nil
-	case IndexAction_Drop:
-		return NewAlterDropIndex(p.db, children[0], p.IndexName), nil
-	case IndexAction_Rename:
-		return NewAlterRenameIndex(p.db, children[0], p.PreviousIndexName, p.IndexName), nil
-	case IndexAction_DisableEnableKeys:
-		return NewAlterDisableEnableKeys(p.db, children[0], p.DisableKeys), nil
+	case IndexAction_Create, IndexAction_Drop, IndexAction_Rename, IndexAction_DisableEnableKeys:
+		p.Table = children[0]
+		return &p, nil
 	default:
 		return nil, ErrIndexActionNotImplemented.New(p.Action)
 	}
 }
 
+func (p AlterIndex) WithTargetSchema(schema sql.Schema) (sql.Node, error) {
+	p.targetSchema = schema
+	return &p, nil
+}
+
+func (p *AlterIndex) TargetSchema() sql.Schema {
+	return p.targetSchema
+}
+
+// Expressions on the AlterIndex object are specifically column default expresions, nothing else.
+func (p *AlterIndex) Expressions() []sql.Expression {
+	newExprs := make([]sql.Expression, len(p.TargetSchema()))
+	for i, col := range p.TargetSchema() {
+		if col.Default != nil {
+			newExprs[i] = expression.WrapExpression(col.Default)
+		} else {
+			newExprs[i] = expression.WrapExpression(nil)
+		}
+	}
+
+	return newExprs
+}
+
+// Update column defaults on the targetSchema instance - required to be the same number of columns on the target schema.
+func (p AlterIndex) WithExpressions(expressions ...sql.Expression) (sql.Node, error) {
+	columns := p.TargetSchema()
```

Comment:
The code below modifies the original schema slice. It should make a copy of it instead. Use `Schema.Copy()`

Reply:
This happens because even though the receiver is a struct, the slice it contains is a reference type, so modifying it modifies the original struct.

Reply:
fixed.